client/jetpack-cloud/sections/partner-portal/company-details-form/index.tsx

```diff
@@ -134,23 +158,38 @@ export default function CompanyDetailsForm( {
 				<FormFieldset>
 					<FormLabel>{ translate( 'Country' ) }</FormLabel>
 					{ showCountryFields && (
-						<SelectDropdown
+						<SearchableDropdown
 							className="company-details-form__dropdown"
-							initialSelected={ country }
+							value={ country }
 							options={ countryOptions }
-							onSelect={ ( option: any ) => {
-								setCountry( option.value );
+							isDisabled={ isLoading }
+							isLoading={ countryOptions.length === 0 }
```

Comment:
I'm not sure if the "disabled/loading" state is working as expecting. Personally I would expect the field to be greyed out or something similar while it's being loaded but we get a `- Select Country -` value in the field instead like the field is empty:

![Screenshot 2022-10-07 at 15 34 08](https://user-images.githubusercontent.com/3846700/194567190-23bc6d0d-18d3-407c-ae73-e7e1c874e5b3.jpg)

And then, when it's done loading, the fields with the existing values are shown:

![Screenshot 2022-10-07 at 15 34 15](https://user-images.githubusercontent.com/3846700/194567275-393e98aa-afb7-4f92-98ad-5df7c07f53b5.jpg)

Reply:
You're right. The reason is the `countryOptions.length` will never be 0, because it'll always have at least one option, the `- Select Country -` option. I let it this way because it was what we had before. I just fixed it, and now the select input will be disabled while it loads the list of countries and will also show a loading indicator on its right.

![image](https://user-images.githubusercontent.com/6406071/198339855-c1822367-34a6-4079-9e2b-75bd6f03ca71.png)

Reply:
You're right. The reason for this is the `countryOptions.length` will never 0, since it's has always the option `- Select Country -`. I just fixed the code to deal with this.

<img width="1254" alt="image" src="https://user-images.githubusercontent.com/6406071/198341573-25dd2bca-89ad-42dc-bdd5-8124c2f0d02e.png">

This is what the users will see while the countries are fetched.

![image](https://user-images.githubusercontent.com/6406071/198342652-b0d0cc88-888d-44ee-9a63-b62e05f7ba51.png)

This was the behavior expected, but it wasn't working because of the bug I found. It's relevant to notice that the list of countries is fetched only once and then cached.